pkg/daemon/ceph/osd/volume.go

```diff
@@ -524,23 +524,36 @@ func (a *OsdAgent) useRawMode(context *clusterd.Context, pvcBacked bool) (bool,
 		useRawMode = false
 	}
 
-	// ceph-volume raw mode does not support more than one OSD per disk
-	osdsPerDeviceCountString := sanitizeOSDsPerDevice(a.storeConfig.OSDsPerDevice)
-	osdsPerDeviceCount, err := strconv.Atoi(osdsPerDeviceCountString)
-	if err != nil {
-		return false, errors.Wrapf(err, "failed to convert string %q to integer", osdsPerDeviceCountString)
-	}
-	if osdsPerDeviceCount > 1 {
-		logger.Debugf("won't use raw mode since osd per device is %d", osdsPerDeviceCount)
-		useRawMode = false
+	// gather node and device config values to verify is there anything to disable raw mode
+	osdsPerDeviceValues := []int{a.storeConfig.OSDsPerDevice}
+	metadataDeviceValues := []string{a.metadataDevice}
+	for _, device := range devices.Entries {
+		osdsPerDeviceValues = append(osdsPerDeviceValues, device.Config.OSDsPerDevice)
+		metadataDeviceValues = append(metadataDeviceValues, device.Config.MetadataDevice)
 	}
 
-	// ceph-volume raw mode mode does not support metadata device if not running on PVC because the user has specified a whole device
-	if a.metadataDevice != "" {
-		logger.Debugf("won't use raw mode since there is a metadata device %q", a.metadataDevice)
-		useRawMode = false
+	// ceph-volume raw mode does not support more than one OSD per disk, verify node and device configs
+	for _, osdsPerDevice := range osdsPerDeviceValues {
+		osdsPerDeviceCountString := sanitizeOSDsPerDevice(osdsPerDevice)
+		osdsPerDeviceCount, err := strconv.Atoi(osdsPerDeviceCountString)
+		if err != nil {
+			return false, errors.Wrapf(err, "failed to convert string %q to integer", osdsPerDeviceCountString)
+		}
+		if osdsPerDeviceCount > 1 {
+			logger.Debugf("won't use raw mode since osd per device is %d", osdsPerDeviceCount)
+			useRawMode = false
```

Comment:
it looks like we need to return list of devices which can be used in raw mode and which can't be used in raw mode. Otherwise we may have situation to set useRawMode for bunch of devices, because of only one device with >1 osd per device. Same for metadata and encrypted devices.